bindings/java/examples/java-app/src/test/java/org/iota/client/example/LibraryTest.java

```diff
@@ -15,51 +18,124 @@ public class LibraryTest {
 
     private static final String NODE = ExampleApp.NODE;
 
+    private Client iota;
+
+    @Before
+    public void init() {
+        iota = node();
+    }
+
     @Test
     public void testNodeInfo() {
-        ExampleApp.nodeInfo();
+        NodeInfoWrapper info = iota.getInfo();
+
+        assertEquals(info.url(), NODE);
+        assertNotNull(info.nodeInfo());
     }
 
     @Test
+    public void testIndexable() {
+        String index = "Hello " + new Date().toString();
+        String data = "Iota.rs java test";
+        Message message1 = iota.message().withIndexString(index).withDataString(data).finish();
+
+        // Fetch message Ids
+        MessageId[] fetched_message_ids = iota.getMessage().indexString(index);
+        assertTrue(fetched_message_ids.length == 1);
+        assertEquals(fetched_message_ids[0], message1.id());
+
+        // get message data
+        Message message1Remote = iota.getMessage().data(fetched_message_ids[0]);
+        assertTrue(message1Remote.payload().isPresent());
+        verifyPayload(message1Remote.payload().get().asIndexation(), index, data);
+
+        // Make index from raw index payload
+        Message message2 = iota.message().finishIndex(IndexationPayload.fromStrings(index, data));
+        // Fetch message Ids
+        fetched_message_ids = iota.getMessage().indexString(index);
+        assertTrue(fetched_message_ids.length == 2);
```

Comment:
This could change at any time

Reply:
thats why the index has a date in it. unless you run it twice in a second or 2 people run it separately simultaneously :)

Reply:
AH i did notice the order isnt deterministic, so i removed the ID equals check